core/domain/exp_services.py

```diff
@@ -792,6 +793,18 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    old_states_dicts = copy.deepcopy(exploration_model.states)
```

Comment:
@prasanna08 I sort of reduced the number of lines in this part of the code. Could you check if we still need a separate function for this? Done with the check for enabling of classifiers.